numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -420,7 +420,7 @@ PyArray_GetStridedCopyFn(int aligned, npy_intp src_stride,
         /* contiguous dst */
         if (itemsize != 0 && dst_stride == itemsize) {
             /* contiguous src */
-            if (itemsize != 0 && src_stride == itemsize) {
```

Comment:
I think the thinking is that `itemsize != 0 && src_stride == itemsize` is the canonical way to spell `is_contiguous`, even if it's repetitive - the compiler can optimize anyway, so the old way had no cost.

It might be better to have an initial condition for `if (itemsize == 0)`, which eliminates these checks from below

Reply:
@eric-wieser Thinking the same. It would also make the logic easier to follow.

Reply:
refactored